projects/views.py

```diff
@@ -484,6 +494,36 @@ def edit_nickname(request, project_id):
     return form
 
 
+
+@require_POST
+def edit_type(request, project_id):
+    form_args = {"request": request}
+    if not request.user.is_superuser:
+        messages.error(request, "Only the admin users can update project type.")
+        return EditTypeForm(**form_args)
+
+    form = EditTypeForm(request.POST, **form_args)
+    if form.is_valid(request):
+        # try to update type
+        try:
+            project_type_id = int(form.cleaned_data["typeId"])
+            ProjectAllocationMapper.update_project_type(project_id, project_type_id)
+            form = EditTypeForm(**form_args)
+            messages.success(
+                request,
+                "Update successful! Click <a href='{}'>here</a> to reload".format(
+                    request.path
+                ),
+            )
+        except Exception:
+            messages.error(request, "Failed to update project type")
```

Comment:
I would also log this exception, otherwise it will be hard to know what happened. You can use `LOG.exception` for that.